Tests/MapboxMapsTests/MapView/Integration Tests/OfflineManagerIntegrationTests.swift

```diff
@@ -0,0 +1,163 @@
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        tileRegionLoadOptions = nil
+    }
+
+    // MARK: Test Cases
+
+    internal func testProgressAndCompletionBlocksBaseCase() {
+        /// Expectations to be fulfilled
+        let downloadInProgress = XCTestExpectation(description: "Downloading offline tiles in progress")
+        downloadInProgress.assertForOverFulfill = false
+        let completionBlockReached = XCTestExpectation(description: "Checks that completion block closure has been reached")
+
+        /// Perform the download
+        TileStore.getInstance().loadTileRegion(forId: tileRegionId,
+                                               loadOptions: tileRegionLoadOptions!) { _ in
+            DispatchQueue.main.async {
+                downloadInProgress.fulfill()
+            }
+        } completion: { result in
+            switch result {
+            case .success(let region):
+                if region.requiredResourceCount == region.completedResourceCount {
+                    completionBlockReached.fulfill()
+                } else {
+                    XCTFail("Not all items were loaded")
+                }
+            case .failure(let error):
+                XCTFail("Download failed with error: \(error)")
+            }
+        }
+
+        let expectations = [downloadInProgress, completionBlockReached]
+        wait(for: expectations, timeout: 5.0)
+    }
+
+    internal func testProgressCanBeCancelled() {
+
+        /// Expectations to be fulfilled
+        let downloadWasCancelled = XCTestExpectation(description: "Checks a cancel function was reached and that the download was canceled")
+
+        /// Perform the download
+        let download = TileStore.getInstance().loadTileRegion(forId: tileRegionId,
+                                                              loadOptions: tileRegionLoadOptions!) { _ in }
+        completion: { result in
+            switch result {
+            case .success(_):
+                XCTFail("Result reached success block, therefore download was not canceled")
+            case .failure(let error):
+                if error.localizedDescription == "Load was canceled" {
+                    downloadWasCancelled.fulfill()
+                } else {
+                    XCTFail("Download was not canceled")
+                }
+            }
+        }
+
+        download.cancel()
+
+        let expectations = [downloadWasCancelled]
+        wait(for: expectations, timeout: 5.0)
+    }
+
+    internal func testOfflineRegionCanBeDeleted() {
+
+        /// Expectations to be fulfilled
+        let downloadWasDeleted = XCTestExpectation(description: "Downloaded offline tiles were deleted")
+
+        /// Perform the download
+        TileStore.getInstance().loadTileRegion(forId: tileRegionId,
+                                               loadOptions: tileRegionLoadOptions!) { _ in } completion: { _ in }
```

Comment:
Let's wait for the `loadTileRegion` to complete using the completion block rather than an `asyncAfter`. We can use another expectation for that.